go/pkg/client/cas.go

```diff
@@ -812,24 +813,46 @@ func (c *Client) ReadBlobStreamed(ctx context.Context, d digest.Digest, w io.Wri
 }
 
 type writerTracker struct {
-	io.Writer
-	writtenBytes int64
+	w     io.Writer
+	pw    *io.PipeWriter
```

Comment:
I wonder if you could avoid the extra goroutine by using an io.MultiWriter instead of io.PipeWriter here?

Reply:
I don't think I could avoid it using a io.MultiWriter the problem was how I set up the `digest.NewFromReader` to run synchronously. While I could change it to be able to stream it, creating a new struct in `digest.go` (to keep encoding code out of `cas.go`) & testing it is some work, and I think there are more important optimizations to do here, eg not close/open a new encoder for every blob.